### PR TITLE
[microTVM] Make Arduino API server obey timeout

### PR DESCRIPTION
**Pull request by guberti**
This change makes the Arduino `read_transport` and `write_transport` methods obey the timeout parameters passed into them. This change is required if we wish to run large models multiple times to get a better average runtime. Also contains a few fixes to improve autotuning:
- Adds an optional `timeout_override` parameter to `create_aot_session`, allowing this function to be used to evaluate large models.
- Changes `tvm.micro.testing.get_target` to use `-device=arm_cpu` by default. 

cc @alanmacd @gromero @mehrdadh

### CODE REVIEW

File: apps/microtvm/arduino/template_project/microtvm_api_server.py [hunk 1]
```diff
@@ -522,13 +522,13 @@ def close_transport(self):
...
     def read_transport(self, n, timeout_sec):
-        # It's hard to set timeout_sec, so we just throw it away
-        # TODO fix this
+        self._serial.timeout = timeout_sec
         if self._serial is None:
             raise server.TransportClosedError()
         return self._serial.read(n)
 
     def write_transport(self, data, timeout_sec):
+        self._serial.timeout = timeout_sec
```
Comment (areusch) [inline apps/microtvm/arduino/template_project/microtvm_api_server.py:531]: i think you should be setting write_timeout, right?

File: python/tvm/micro/testing/evaluation.py [hunk 2]
```diff
@@ -83,6 +83,7 @@ def create_aot_session(
     params,
     build_dir=Path(tempfile.mkdtemp()),
     tune_logs=None,
+    timeout_override=None,
```
Comment (areusch) [inline python/tvm/micro/testing/evaluation.py:86]: nit: units e.g. timeout_sec_override

**Review by areusch — COMMENTED** [review]
few small things, lgtm otherwise!

File: apps/microtvm/arduino/template_project/microtvm_api_server.py (hunk 1 above)
Comment (guberti) [reply apps/microtvm/arduino/template_project/microtvm_api_server.py:531]: Nice catch!

File: python/tvm/micro/testing/evaluation.py (hunk 2 above)
Comment (guberti) [reply python/tvm/micro/testing/evaluation.py:86]: This isn't a float - this should be a `tvm.micro.project_api.server.TransportTimeouts` object (`timeout_override` is always this type in our codebase). We could add typing hints to this file, but I think that's OOS for this PR.

**Review by alanmacd — APPROVED** [review]

**Pull request merged**